docs/explanation/sitemaps.md

```diff
@@ -0,0 +1,225 @@
+# Sitemaps in WordPress: A Comprehensive Overview
+
+## What is a Sitemap?
+A sitemap is an XML file that provides a structured list of pages on a website by helping search engines discover and crawl content more efficiently. It acts as a roadmap of your website's structure, containing important metadata about each page.
+Since WordPress 5.5, there's a built-in XML sitemap generator that:
+
+* Automatically creates sitemaps for posts, pages, and custom post types
+* Dynamically updates as content is added, modified, or deleted
+* Provides basic SEO and indexing support out of the box
+
+However, this default sitemap lacks customization options, which is why many users opt for plugins like Yoast SEO or Jetpack to generate more comprehensive sitemaps.
+
+You can view the current sitemap generated by WordPress by visiting yourdomain.com/sitemap.xml in your browser.
+![Image showing sitemap for a specific website](./sitemaps-1.png)
+
+## Sitemaps in Headless WordPress
+Headless WordPress environments introduce unique challenges for sitemap generation:
+
+* **Separation of content management and frontend rendering**: In traditional WordPress setups, content management and frontend rendering are integrated. However, in headless environments, WordPress acts as a backend CMS, while the frontend is handled by frameworks like Next.js. This separation requires custom solutions for sitemap generation.
+
+* **Dynamic routes**: Dynamic routes in frameworks like Next.js may not be directly accessible or easily discoverable by search engines.
+
+* **Need for custom sitemap generation and management**: Due to the decoupling of backend and frontend, traditional WordPress sitemap plugins might not work seamlessly. Therefore, custom approaches are needed to generate and manage sitemaps effectively.
+
+To address those, there are some proposed solutions for headless sitemap implementation:
+
+1. **Proxying Sitemaps from Backend to Frontend**
+
+Approach: This approach maintains WordPress's native sitemap generation capabilities while ensuring proper frontend URLs are used. It involves creating a route in your frontend application that proxies requests to the WordPress backend sitemap.
+
+Example Code:
+
+```javascript
+// pages/[...sitemap].js
+export async function GET(request, { params }) {
+  const { sitemap } = params;
+  const sitemapFile = Array.isArray(sitemap) ? sitemap.join('/') : sitemap;
+  
+  const wpSitemapUrl = `${process.env.WORDPRESS_URL}/wp-sitemap${sitemapFile ? `-${sitemapFile}` : ''}.xml`;
```

Comment:
I don't really understand this code. A catch-all route wouldn't work with most other routing situations. Aditionally, any unknown URL that should 404 would just render a sitemap.